src/kOS/Screen/TermWindow.cs

```diff
@@ -625,21 +625,36 @@ public void ProcessOneInputChar(char ch, TelnetSingletonServer whichTelnet)
 
         void Type(char ch)
         {
-            if (shared != null && shared.Interpreter != null)
+            if (shared != null)
             {
-                shared.Interpreter.Type(ch);
-                if (IsOpen && keyClickEnabled)
-                    shared.SoundMaker.BeginSound("click");
+                if (shared.Interpreter != null && shared.Interpreter.isWaitingForCommand())
+                {
+                    shared.Interpreter.Type(ch);
+                    if (IsOpen && keyClickEnabled)
+                         shared.SoundMaker.BeginSound("click");
```

Comment:
Ah darn.  I just realized the way this is implemented, the click only happens when sending the key to the interpreter to process.  This clicker gets skipped when sending the key to a user program to process in the else clause below.

I'm pretty sure very few people are using the key clicker, but for the few who are, I should probably change this.  the key click is supposed to be a feature of the terminal itself, not of which program is running inside it.